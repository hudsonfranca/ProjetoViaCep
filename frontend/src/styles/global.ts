import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
*{
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Rubik', sans-serif;
    outline: none;
    text-decoration:none;
}
  body {
    min-height: 100vh;
  }
`