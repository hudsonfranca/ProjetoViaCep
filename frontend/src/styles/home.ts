import styled from "styled-components";


export const Container = styled.div`
  min-height: 100%;
  width: 100%;
  background-color:${({theme:{colors:{primary}}})=>primary};
  padding-top:100px;
  padding-bottom:100px;
`

export const Section = styled.section`
  min-height: 100vh;
  width: 100%;
  background-color:${({theme:{colors:{primary}}})=>primary};
  color:#fff;
  padding: 0 10%;
  display:flex;
  flex-direction:column;
  align-items: center;

  h1{
    font-size:2.5rem;
    margin-bottom:30px;
  }

  input{
    width: 100%;
    height: 100%;
    padding: 4px 14px;
    font-size:1.7rem;
    border: 0;
    color:#495057;
   
  }

  @media (max-width: 413px) {
    padding: 0 7%;

    input{
      font-size:1.5rem;
    }
  }
  
`

export const Form = styled.form`
  width: 100%;
  height: 81px;
  background-color:#fff;
  padding: 10px 21px;
  border-radius:0.25rem;
  display: grid;
  grid-template-columns:1fr auto;
  justify-content: center;
  align-items: center;
`

export const Button = styled.button`
  width: 111px;
  height: 61px;
  color: #fff;
  background-color:#20c997;
  border: 0 solid #20c997;
  font-size:1rem;
  font-weight:500;
  border-radius:0.25rem;
  cursor: pointer;
  transition:0.3s ease-in-out;
  display:flex;
  justify-content: center;
  align-items: center;

  &:hover{
  background-color:#02b875;
  }
  @media (max-width: 413px) {
    width: 100px;
  height: 49px
  }

`

