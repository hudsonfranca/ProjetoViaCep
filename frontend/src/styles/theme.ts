export const theme = {
  colors: {
    primary: '#4582ec',
    secondary:'#343a40',
    tertiary:'#44b976',
    quaternary:'#F0C10E'
  },
}
