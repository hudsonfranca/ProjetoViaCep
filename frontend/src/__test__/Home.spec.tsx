
/**
 * @jest-environment jsdom
 */

 import { render, screen } from '@testing-library/react';
 import { ThemeProvider } from 'styled-components';
 import Home from '../pages';
 import { theme } from '../styles/theme';
 import 'jest-styled-components'

 const renderComponent = (props = {}) => {
   return {
     ...render(
     <ThemeProvider theme={theme}>
       <Home/>
     </ThemeProvider>
   )
   };
 };
 
describe('<Home/>',()=>{

  it('<Home/> Snapshot test', () => {
    const {container} = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('must render <Home/>',()=>{
  
    renderComponent()
    expect(screen.getByRole('heading', {
      name: /buscar código de endereçamento postal/i
    })).toBeInTheDocument()
  
    expect(screen.getByRole('searchbox', {
      name: /buscar cep/i
    })).toBeInTheDocument()
  
    expect(screen.getByRole('button', {
      name: /buscar/i
    })).toBeInTheDocument()
    
  })
})