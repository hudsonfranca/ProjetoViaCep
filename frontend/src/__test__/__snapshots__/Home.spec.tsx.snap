// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> <Home/> Snapshot test 1`] = `
.c0 {
  min-height: 100%;
  width: 100%;
  background-color: #4582ec;
  padding-top: 100px;
  padding-bottom: 100px;
}

.c1 {
  min-height: 100vh;
  width: 100%;
  background-color: #4582ec;
  color: #fff;
  padding: 0 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 h1 {
  font-size: 2.5rem;
  margin-bottom: 30px;
}

.c1 input {
  width: 100%;
  height: 100%;
  padding: 4px 14px;
  font-size: 1.7rem;
  border: 0;
  color: #495057;
}

.c2 {
  width: 100%;
  height: 81px;
  background-color: #fff;
  padding: 10px 21px;
  border-radius: 0.25rem;
  display: grid;
  grid-template-columns: 1fr auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 111px;
  height: 61px;
  color: #fff;
  background-color: #20c997;
  border: 0 solid #20c997;
  font-size: 1rem;
  font-weight: 500;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:hover {
  background-color: #02b875;
}

@media (max-width:413px) {
  .c1 {
    padding: 0 7%;
  }

  .c1 input {
    font-size: 1.5rem;
  }
}

@media (max-width:413px) {
  .c3 {
    width: 100px;
    height: 49px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="Toastify"
    />
    <section
      class="c1"
    >
      <h1>
        Buscar 
        <abbr
          title="Código de Endereçamento Postal"
        >
          CEP
        </abbr>
      </h1>
      <form
        class="c2"
        role="search"
      >
        <input
          aria-label="Buscar CEP"
          aria-required="true"
          id="cep"
          name="cep"
          type="search"
          value=""
        />
        <button
          aria-label="Buscar"
          class="c3"
          type="submit"
        >
          Buscar
        </button>
      </form>
    </section>
  </div>
</div>
`;
